src/Core/Language/Pack/Import/LanguagePackImporter.php

```diff
@@ -0,0 +1,125 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Language\Pack\Import;
+
+use Exception;
+use PrestaShop\PrestaShop\Adapter\Cache\CacheClearer;
+use PrestaShop\PrestaShop\Adapter\Language\LanguageDataProvider;
+use PrestaShop\PrestaShop\Adapter\Language\LanguagePack;
+use PrestaShop\PrestaShop\Core\Cldr\Update;
+
+/**
+ * Class LanguagePackImporter is responsible for importing language pack
+ */
+final class LanguagePackImporter implements LanguagePackImporterInterface
+{
+    /**
+     * @var LanguagePack
+     */
+    private $languagePack;
+    /**
+     * @var LanguageDataProvider
+     */
+    private $languageProvider;
+    /**
+     * @var CacheClearer
+     */
+    private $cacheClearer;
+
+    /**
+     * @param LanguagePack $languagePack
+     * @param LanguageDataProvider $languageProvider
+     * @param CacheClearer $cacheClearer
+     */
+    public function __construct(
+        LanguagePack $languagePack,
+        LanguageDataProvider $languageProvider,
+        CacheClearer $cacheClearer
```

Comment:
it heavily depends on adapters, maybe class should be moved to adapter namespace?